### DISCUSSION TITLE
[Assignment 2]: Costmap doesn't include the new sensor data

### DISCUSSION THREAD

**Discussion by erodios** [gstavrinos/ai_msc_robotic_systems_assignments#13] [Q&A]
I have included in the local and global costmap one new sensor (camera) and there are some issues that i can't figure out. So sometimes the table will apear in the costmap and some other times it will not, and when it apears and the robot move the rest of the table wont be updated in the costmap so all that the robot sees is only half the table that it could see at the start. 

I don't know if it is a problem with humble cause i found this thread (it is about Costmap Topic Collision Checker ) " https://discourse.ros.org/t/nav2-issues-with-humble-binaries-due-to-fast-dds-rmw-regression/26128 " and they suggest to switch to Cyclone DDS. 

Should i try this solution?

**Comment by gstavrinos** [comment]
This is a known Fast-DDS bug as you mentioned, but has been, presumably, patched. Either way, it is very easy to switch to Cyclone DDS, so I would suggest giving it a try. Also, updating your packages (`sudo apt update && sudo apt full-upgrade -y`) wouldn't hurt, just to be sure.

Please report back if this was indeed the problem.

I would assume that the problem lies on the ROS2-side and not yours. In any case, make sure that your configuration allows for costmap updates, that your sensors actually see the expected obstacles and that some parameter is not "filtering out" new obstacles.

**Reply by alecfilios** [reply]
I face the same issue, I tried several ways of parameter tuning in the configuration file. Are there any solutions that will enable a local cost map frequent update?

**Comment by gstavrinos** [comment]
A new announcement has been posted on e-class, regarding this issue.

In order to help you avoid having to tackle this specific issue, please consider using the newly added _easy_mode_.

Any feedback or potential fixes are welcome.

**Reply by alecfilios** [reply]
Any hints for those who want to avoid the easy-mode and go with PointCloud2?

**Reply by alecfilios** [reply]
Basically, our problem is that local-cost map using point cloud2 does not update at all. The only times it updates is when the robot falls or hits an obstacle. Thus, after the first goal it tries to fgo through the table. 

Apart from that, at the end of the journey to the first goal, it spins and tries to position itself for a long time. I do not know if this is something normal.

**Reply by erodios** [reply]
[Screencast from 06-25-2023 06:10:15 PM.webm](https://github.com/gstavrinos/ai_msc_robotic_systems_assignments/assets/56002896/4104527c-f46d-405c-9760-6514b91fbd7e)
So i tried to use the easy_mode and the robot had this behavior when it rotates next to this part of the table,
 is this normal and i haave to change some parameters to fix it? (The robot didn't move in Gazebo)

[Screencast from 06-25-2023 07:23:37 PM.webm](https://github.com/gstavrinos/ai_msc_robotic_systems_assignments/assets/56002896/83c23756-6f46-41ed-ac3d-d56edc2f6fa5)
in the second video is the issue where the costmap is not updated from the pointcloud after the initialization

**Reply by alecfilios** [reply]
I have the exact same problems with both approaches. With many variations of parameter tuning.

**Comment by gstavrinos** [comment]
> [Screencast from 06-25-2023 06:10:15 PM.webm](https://github.com/gstavrinos/ai_msc_robotic_systems_assignments/assets/56002896/4104527c-f46d-405c-9760-6514b91fbd7e) So i tried to use the easy_mode and the robot had this behavior when it rotates next to this part of the table, is this normal and i haave to change some parameters to fix it? (The robot didn't move in Gazebo)
> 
> [Screencast from 06-25-2023 07:23:37 PM.webm](https://github.com/gstavrinos/ai_msc_robotic_systems_assignments/assets/56002896/83c23756-6f46-41ed-ac3d-d56edc2f6fa5) in the second video is the issue where the costmap is not updated from the pointcloud after the initialization



> I have the exact same problems with both approaches. With many variations of parameter tuning.


A fix for this issue has been pushed. Pull the [ai_msc_robotic_systems_assignments](https://github.com/gstavrinos/ai_msc_robotic_systems_assignments) and the [gazebo_table_tennis](https://github.com/gstavrinos/gazebo_table_tennis) repos to get it. Sorry for that! :disappointed_relieved:

**Reply by alecfilios** [reply]
It worked for me, thank you so much!
